pkg/azuredisk/nodeserver_v2.go

```diff
@@ -234,9 +228,9 @@ func (d *DriverV2) NodePublishVolume(ctx context.Context, req *csi.NodePublishVo
 
 	switch req.GetVolumeCapability().GetAccessType().(type) {
 	case *csi.VolumeCapability_Block:
-		lunStr, ok := req.PublishContext[consts.LUN]
-		if !ok {
-			return nil, status.Error(codes.InvalidArgument, "lun not provided")
+		lunStr, err := d.nodeProvisioner.GetLunFromVolume(volumeID)
```

Comment:
Since the node instance could crash and restart between `NodeStageVolume` and `NodePublishVolume` we can't really rely on cached state being valid between the two. It would be safer to provide a means to get the publish context for a given disk URI from the corresponding `AzVolumeAttachment` through the `CrdProvisioner`, e.g.:
```go
func (p *CrdProvisioner) GetPublishContext(ctx context.Context, diskURI, nodeID string) (map[string]string, error)
```

Reply:
Ah yes. In that case, we would need to rely on the `AzVolumeAttachment` object to get the necessary information as you pointed out. Until we have the informer change implemented in `CrdProvisioner`, I presume this would add a bit of overhead as it would require a `Get` from the api-server.